### ISSUE TITLE
"Qml library not loaded" error with PySide6 application

### ISSUE THREAD

**Issue by Merkwurdichliebe** [Nuitka/Nuitka#1195]
When compiling a Python script which uses PySide6 on macOS (Catalina 10.15.7), I get the following error:

```
ImportError: dlopen(/Users/me/myapp/barebones.dist/./PySide6/QtWidgets.so, 2): Library not loaded: @rpath/QtQml.framework/Versions/A/QtQml
  Referenced from: /Users/me/myapp/barebones.dist/libpyside6.abi3.6.1.dylib
  Reason: image not found
```

This happens with my real app, which isn't very complex, but I've confirmed the same result with the following single-file code:

```
if (__name__ == '__main__'):
    from PySide6.QtWidgets import QApplication
    from PySide6.QtWidgets import QGraphicsScene, QGraphicsView
    a = QApplication()
    s = QGraphicsScene()

    v = QGraphicsView(s)
    v.show()
    a.exec()
```

I am not using QML at all.

I've tried both with and without this option: `--plugin-enable=pyside6`.

* Nuitka version, full Python version and Platform (Windows, OSX, Linux ...)

0.6.16.4
Python: 3.9.6 (default, Jun 29 2021, 06:20:32) 
Executable: /Users/me/myapp/.venv/bin/python3
OS: Darwin
Arch: x86_64

* How did you install Nuitka and Python

`pip3 install nuitka` from within a virtual environement

* The specific PyPI names and versions

  It should be taken from this output if there specific packages involved, e.g.
  `numpy`, you are expected to shorten this to the relevant ones.

PySide6==6.1.2
shiboken6==6.1.2

* Provide in your issue the Nuitka options used

` python3 -m nuitka --standalone --plugin-enable=pyside6 barebones.py`

* Note if this is a regression

First-time user.

**Comment by kayhayen** [comment]
Yes, this is a problem, specific to macOS, which seems to come from macOS standalone freezer code scanning for DLLs not having the mechanism to provide the framework directories during its detection. Both Windows and Linux (POSIX systems, but not macOS) have gained consideration of it, but macOS needs to be done still.

**Comment by kayhayen** [comment]
This is part of the current stable release, but not that for macOS standalone to be good, you will need 0.6.18 and should use develop branch.

**Comment by zoell** [comment]
@kayhayen quick question: Is this fixed in any Nuitka version? What do you mean by use 0.6.18 and should use develop branch? I am currently trying with 0.6.19.3 and the issue is there (error is exactly the same as above).

**Comment by kayhayen** [comment]
It is supposed to work for as long as you have used the plugin, yes.

**Comment by zoell** [comment]
That is interesting, but I am using the version installed by pip. Is that also updated now or only the develop branch?

**Comment by kayhayen** [comment]
It should work, if it doesn't, raise a new issue.

**Comment by zoell** [comment]
OK so I tested the scenario that was mentioned above by @Merkwurdichliebe (with the sample code and Nuitka command provided), same error message (not sure if it worth to open a new bug as it is exactly the same bug):

```
File "/Users/myuser/Dev/test/test.dist/test.py", line 2, in <module>
ImportError: dlopen(/Users/myuser/Dev/test/test.dist/PySide6/QtWidgets.so, 0x0002): Library not loaded: @rpath/QtQml.framework/Versions/A/QtQml
Referenced from: /Users/myuser/Dev/test/test.dist/libpyside6.abi3.6.2.dylib
Reason: tried: '/Users/myuser/Dev/test/test.dist/Qt/lib/QtQml.framework/Versions/A/QtQml' (no such file), '$ORIGIN/QtQml.framework/Versions/A/QtQml' (no such file), '/Users/myuser/Dev/test/test.dist/PySide6/Qt/lib/QtQml.framework/Versions/A/QtQml' (no such file), '$ORIGIN/../QtQml.framework/Versions/A/QtQml' (no such file), '/Users/myuser/Dev/test/test.dist/PySide6/Qt/lib/QtQml.framework/Versions/A/QtQml' (no such file), '$ORIGIN/../QtQml.framework/Versions/A/QtQml' (no such file), '$ORIGIN/QtQml.framework/Versions/A/QtQml' (no such file), '$ORIGIN/QtQml.framework/Versions/A/QtQml' (no such file), '/Library/Frameworks/QtQml.framework/Versions/A/QtQml' (no such file), '/System/Library/Frameworks/QtQml.framework/Versions/A/QtQml' (no such file)Library not loaded: @rpath/QtDBus.framework/Versions/A/QtDBus
```

OS: macOS Monterey 12.1
Nuitka: 0.6.19.3
Python: 3.9.10
PySide6: 6.2.2.1

**Comment by kayhayen** [comment]
Doesn't look like it to me though, absolutely not the same @zoell but I can imagine that an upgrade of PySide has introducing something that confuses the parser in nuitka. That `@rpath` should not survive standalone handling. Just because a symptom is the same, doesn't mean it's the same issue. The program is going to break over and over in new ways over the years I guess.

**Comment by kayhayen** [comment]
I am locking the issue, to avoid incomplete "me too" reports like yours, and have no issues created.
